sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -105,12 +107,28 @@ abstract class ColumnarOutputWriter(path: String, context: TaskAttemptContext,
    */
   def write(batch: ColumnarBatch, statsTrackers: Seq[ColumnarWriteTaskStatsTracker]): Unit = {
     var needToCloseBatch = true
+    val outputTimestampType = conf.get(SQLConf.PARQUET_OUTPUT_TIMESTAMP_TYPE.key)
+    val newBatch = if (outputTimestampType == ParquetOutputTimestampType.TIMESTAMP_MILLIS) {
```

Comment:
Does this work?  It looks like we're comparing a string with an enum here.

Reply:
Thank you for pointing this out!